bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/GC.java

```diff
@@ -133,6 +134,13 @@ public final class GC extends Resource {
  */
 public GC(Drawable drawable) {
 	this(drawable, SWT.NONE);
+	if (drawable instanceof Control) {
+		data.setNativeDeviceZoom(((Control) drawable).getShell().nativeZoom);
+	} else if(drawable instanceof Widget) {
+		data.deviceZoom = ((Widget) drawable).zoom;
```

Comment:
Why is it okay here that `nativeDeviceZoom` is unset? This will result in `getDeviceZoom()` returning some fallback value (based on `DPIUtil`). If I see correctly, this is used by the `setFont` method, which then creates a font with a scale value vased on that fallback value. My naive expectation would be that this will eventually fail when having multiple shells on monitors with different scale values and using both Controls (with a properly set native zoom) and Widgets (with a fallback value). Even if that behavior is intended, it is rather incomprehensible why is makes sense to use a fallback value in some cases.